src/tmpfiles/tmpfiles.c

```diff
@@ -1261,7 +1261,9 @@ static int create_item(Item *i) {
                 log_debug("%s directory \"%s\".", creation_mode_verb_to_string(creation), i->path);
 
                 if (IN_SET(i->type, CREATE_SUBVOLUME_NEW_QUOTA, CREATE_SUBVOLUME_INHERIT_QUOTA)) {
-                        r = btrfs_subvol_auto_qgroup(i->path, 0, i->type == CREATE_SUBVOLUME_NEW_QUOTA);
+                        r = path_check_fstype(i->path, BTRFS_SUPER_MAGIC);
+                        if (r == 1)
+                                 r = btrfs_subvol_auto_qgroup(i->path, 0, i->type == CREATE_SUBVOLUME_NEW_QUOTA);
```

Comment:
ioctl's don't work on O_PATH file descriptors.  fstat and fstatfs and a few fcntls() work, and they can be used as the starting point for openat() and friends. Hmm... man page doesn't mention fstatfs. I should fix that.
O_PATH on an autofs mount point will not trigger the mount.  If it has previously been triggered and something is mounted on top of the autofs mount point, then open(O_PATH) will get whatever is on "top".
You could get btrfs_subvol_auto_qgroup() to open with O_PATH, check if it is btrfs, then if it is - openat(fd, ".", O_DIRECTORY|O_RDONLY...) to get an fd for ioctl.  That would avoid many races.... though if some other filesystem was mounted on the directory you could lose.  How much paranoia is useful here?

The reason that I didn't offer code to check everything for autofs is that I think autofs should be as transparent as possible.  In this case, it isn't that we don't want autofs, it is that we *do* want btrfs.  Actually would be be more correct to be testing if "/" is btrfs, and skipping btrfs_subvol_auto_qgroup() in that case?  The documentation for 'v' suggests "/" must be btrfs, and 'q' and 'Q'  are based on 'v'.  Would it then be correct to test if i->path is the same filesystem as "/" ???